kratos/spatial_containers/bins_dynamic_objects.h

```diff
@@ -107,21 +108,25 @@ class BinsObjectDynamic {
   BinsObjectDynamic (IteratorType const& ObjectsBegin, IteratorType const& ObjectsEnd)
       : mObjectsBegin(ObjectsBegin), mObjectsEnd(ObjectsEnd) {
 
+    std::size_t GridSize[3] = {0, 0, 0};
+
     mObjectsSize = SearchUtils::PointerDistance(mObjectsBegin,mObjectsEnd);
-    CalculateBoundingBox();           // Calculate mMinPoint, mMaxPoint
-    CalculateCellSize(mObjectsSize);  // Calculate number of Cells
-    AllocateContainer();              // Allocate cell list
-    GenerateBins();                   // Fill Cells with objects
+    CalculateBoundingBox();                                                 // Calculate mMinPoint, mMaxPoint
+    CalculateCellSize(mObjectsSize, GridSize);                              // Calculate number of Cells
+    AllocateContainer();                                                    // Allocate cell list
+    GenerateBins();                                                         // Fill Cells with objects
   }
 
   BinsObjectDynamic (IteratorType const& ObjectsBegin, IteratorType const& ObjectsEnd, CoordinateType CellSize)
```

Comment:
This constructor cannot be used in the terms of the new Calculatesize function, so I've dimension the GridSize as an aproximation of the CellSize provided. Not sure how this works but was the only way to mantain the compativility

Reply:
You may rename the CaluclateCellSize(CellSize) to AssignCellSize and use it here as it is.